### PR TITLE
[FIX] account: properly unlink tags when unlinking a tax report line or setting tag_name to None

### PR DESCRIPTION
**Pull request by oco-odoo**

### CODE REVIEW

File: addons/account/models/account.py [hunk 1]
```diff
@@ -138,16 +137,26 @@ def write(self, vals):
...
 
+    def _remove_tax_used_only_by_self(self):
+        """ Deletes and removes from taxes and move lines all the
+        tags from the provided tax report lines that are not linked
+        to any other tax report lines.
+        """
+        all_tags = self.mapped('tag_ids.id')
+        tags_to_unlink = all_tags.filtered(lambda x: not (x.tax_report_line_ids - self))
+        self.write({'tag_ids': [(2, tag.id, 0) for tag in tags_to_unlink]})
+        self._delete_tags_from_taxes(tags_to_unlink)
```
Comment (smetl) [inline addons/account/models/account.py:154]: Shouldn't be tags written after the deletion?

Comment (smetl) [reply addons/account/models/account.py:154]: Also, in case of unlink, this is strange to write something before unlinking the records :x

Comment (oco-odoo) [reply addons/account/models/account.py:154]: @smetl Strange, but necessary and cleaner written this way: tags can be shared between multiple report lines (in multiple reports); you don't want to delete them every time.

Also, I don't understand your first comment :p

Comment (smetl) [reply addons/account/models/account.py:154]: Forget the first comment.
Can you also add a test plz?

**Comment by smetl** [comment]
@robodoo r+

**Comment by oco-odoo** [comment]
@fw-bot up to 13.0

I'll do the forward-port to v14 in a distinct PR, adding a test in https://github.com/odoo/odoo/blob/14.0/addons/account/tests/test_tax_report.py

**Pull request closed without merging**